go/consensus/tendermint/apps/roothash/state/state.go

```diff
@@ -193,7 +197,16 @@ func NewMutableState(tree mkvs.KeyValueTree) *MutableState {
 
 // SetRuntimeState sets a runtime's roothash state.
 func (s *MutableState) SetRuntimeState(ctx context.Context, state *roothash.RuntimeState) error {
-	err := s.ms.Insert(ctx, runtimeKeyFmt.Encode(&state.Runtime.ID), cbor.Marshal(state))
+	if err := s.ms.Insert(ctx, runtimeKeyFmt.Encode(&state.Runtime.ID), cbor.Marshal(state)); err != nil {
+		return api.UnavailableStateError(err)
+	}
+
+	// Store just the current state root separately to make it easier to retrieve.
+	stateRoot, err := state.CurrentBlock.Header.StateRoot.MarshalBinary()
```

Comment:
Although not needed at the moment, maybe include a getter and a test for retrieving the state root.

Reply:
Done.